src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -171,20 +173,23 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
                 }
             }
 
+            Debug.Assert(!sourceDocuments.IsEmpty);
+            var checksumAlgorithm = sourceDocuments[0].ChecksumAlgorithm;
```

Comment:
Can we be doing something smarter than just grabbing the first one here?

Reply:
The value of /checksumalgorithm is not stored in the PDB, so not really. It doesn't matter much in this case since the project is synthesized project that holds on MAAS documents.